src/Native/CMakeLists.txt

```diff
@@ -4,6 +4,11 @@ project(CoreRT)
 # Include cmake functions
 include(functions.cmake)
 
+string_has_prefix(${CMAKE_GENERATOR} "Visual Studio 15 2017" GENERATOR_IS_VS2017)
```

Comment:
You don't need to introduce a new function. `string(FIND ${CMAKE_GENERATOR} "Visual Studio 15 2017" VS2017_POS)` and comparing VS2017_POS to 0 would work for this purpose.